distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -765,8 +771,34 @@ Text.ends_with suffix = Text_Utils.ends_with this suffix
      See if the text "Hello" contains the text "ell".
 
          "Hello".contains "ell"
-Text.contains : Text -> Boolean
-Text.contains sequence = Text_Utils.contains this sequence
+
+   > Example
+     See if the text "Cześć" contains the text 's\u{301}' (which folds to 'ś').
+
+         "Cześć".contains 's\u{301}'
+
+   > Example
+     See if the text "Hello!" contains the text 'LO', ignoring case sensitivity.
+
+         "Hello!".contains "LO" (Text_Matcher Case_Insensitive.new)
+
+   > Example
+     See if the text "Hello!" contains any lowercase letters, using a regex.
+
+         "Hello!".contains "[a-z]" Regex_Matcher.new
+Text.contains : Text -> (Text_Matcher | Regex_Matcher) -> Boolean
+Text.contains term="" matcher=Text_Matcher.new = case matcher of
+    Text_Matcher case_sensitivity -> case case_sensitivity of
+        True -> Text_Utils.contains this term
+        Case_Insensitive locale ->
+            Text_Utils.contains (this.to_lower_case locale) (term.to_lower_case locale)
+    Regex_Matcher case_sensitive multiline match_ascii dot_matches_newline comments ->
+        case_insensitive = case case_sensitive of
+            True -> False
+            Case_Insensitive _ -> True
```

Comment:
should we do case insensitive the same way as for Text_Matcher?

Reply:
could do that, yeah